[master] Add support for show_jid to salt-run

barneysowood opened the pull request:
### What does this PR do?

Adds support for show_jid to the salt-run cli command to match the behaviour of the salt cli command.

Could have implemented this entirely within `salt/runner.py`, but I think it makes sense to not output the JID when Runner is called from other places - happy to get any feedback or alternative suggests on that.

### Previous Behavior
If `show_jid: True` was set in master config, running `salt-run` would not show a JID.

### New Behavior

`salt-run` now prints `jid: <jid>`

### Merge requirements satisfied?

- [X] Changelog
- [X] Tests written/updated

### Commits signed with GPG?
No

barneysowood committed:
Add support for show_jid to salt-run

Adds support for show_jid to the salt-run cli command to match the
behaviour of the salt cli command.

barneysowood committed:
Add changelog

Ch3LL commented:
Those package tests will continue to fail until our merge forward PR into master is merged. I'll make sure to update the branch once I know when its merged.

barneysowood commented:
> Those package tests will continue to fail until our merge forward PR into master is merged. I'll make sure to update the branch once I know when its merged.

@Ch3LL I'm still seeing CI failures, I take it the merge forward PR is still outstanding?

Thanks!

Ch3LL commented:
The package tests are passing now. Let me restart the failing ones.

barneysowood commented:
re-run all

mattp- commented on salt/cli/run.py:
```diff
@@ -30,8 +30,9 @@ def run(self):
+                show_jid = self.config.get("show_jid", False)
                 try:
-                    ret = runner.run(full_return=True)
+                    ret = runner.run(full_return=True, show_jid=show_jid)
```
why does this need to be passed as a kwarg? salt.runner has self.opts (the same as self.config here)

barneysowood replied:
@mattp- - thanks for the feedback. My intent in doing that was to ensure that the JID was *only* printed when the runner was called from `salt-run`. I'd assumed if I didn't do that, when called via the python api or netapi, that would be either returned or logged.. having tested with your suggested changes however, that doesn't seem to be the case.

I'm not sure where the stdout goes in those cases, but it doesn't seem to be an issue, so your much simpler solution is good - thanks!

mattp- commented on salt/runner.py:
```diff
@@ -288,6 +288,9 @@ def run(self, full_return=False):
                     return async_pub["jid"]  # return the jid
 
                 # otherwise run it in the main process
+                if show_jid:
```
```suggestion
                if self.opts.get('show_jid'):
```

barneysowood committed:
Simplify to just use opt in runner

Simplify to just use opts in runner to get show_jid. Had thought
I'd need to do that so not to print when called via python api or
netapi, but doesn't seem to be an issue.

barneysowood committed:
Merge branch 'master' into show-jid-salt-run
